### PR TITLE
client and securable item entity models, mappers, and db context setup

### PR DESCRIPTION
**Pull request by kpaul**
i need to figure out the best way to model the securableItems collection on the securableItem entity. Its supposed to represent the children securable items. 

i also left a placeholder string for created by and modified by until we have a way to pull the authenticated users name. i will add a task for it.

### CODE REVIEW

File: Fabric.Authorization.Persistence.SqlServer/EntityModels/Client.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using Fabric.Authorization.Domain.Models;
+
+namespace Fabric.Authorization.Persistence.SqlServer.EntityModels
+{
+    public class Client : ITrackable, ISoftDelete
+    {
+        public int Id { get; set; }
+        public string ClientId { get; set; }
+        public string Name { get; set; }
+        public int SecurableItemId { get; set; }
```
Comment (bsmithvt) [inline Fabric.Authorization.Persistence.SqlServer/EntityModels/Client.cs]: should this be nullable, or is it required when creating a client?

File: Fabric.Authorization.Persistence.SqlServer/EntityModels/Client.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Fabric.Authorization.Domain.Models;
+
+namespace Fabric.Authorization.Persistence.SqlServer.EntityModels
+{
+    public class Client : ITrackable, ISoftDelete
+    {
+        public int Id { get; set; }
+        public string ClientId { get; set; }
+        public string Name { get; set; }
```
Comment (bsmithvt) [inline Fabric.Authorization.Persistence.SqlServer/EntityModels/Client.cs]: should this be nullable, or is it required when creating a client?

Comment (kpaul) [reply Fabric.Authorization.Persistence.SqlServer/EntityModels/Client.cs]: we validate it as a required field when posting a new one

Comment (kpaul) [reply Fabric.Authorization.Persistence.SqlServer/EntityModels/Client.cs]: good catch, it should be nullable. i will update that

File: Fabric.Authorization.Persistence.SqlServer/EntityModels/SecurableItem.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        public string Name { get; set; }
+
+        public DateTime CreatedDateTimeUtc { get; set; }
+        public DateTime? ModifiedDateTimeUtc { get; set; }
+        public string CreatedBy { get; set; }
+        public string ModifiedBy { get; set; }
+        public bool IsDeleted { get; set; }
+
+        public ICollection<SecurableItem> SecurableItems { get; set; }
+        public ICollection<Client> Clients { get; set; }
```
Comment (michaelvidal24) [inline Fabric.Authorization.Persistence.SqlServer/EntityModels/SecurableItem.cs]: Shouldn't a SecurableItem only be related to a single Client?

Comment (kpaul) [reply Fabric.Authorization.Persistence.SqlServer/EntityModels/SecurableItem.cs]: That is what I was not sure about. I can update the model to reflect the 1:1 relationship if it should be

**Pull request merged**